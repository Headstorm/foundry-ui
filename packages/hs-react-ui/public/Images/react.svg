<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 19.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 width="561.8px" height="499.9px" viewBox="-80.4 151.2 561.8 499.9" style="enable-background:new -80.4 151.2 561.8 499.9;"
	 xml:space="preserve">
<style type="text/css">
	.st0{fill:#939598;}
</style>
<circle class="st0" cx="200.5" cy="400.6" r="50.2"/>
<path class="st0" d="M200.5,515.6c-70.5,0-132.1-8.3-178.2-24.1c-29.9-10.2-55.3-23.8-73.4-39.3c-19.2-16.4-29.4-34.3-29.4-51.6
	c0-33.2,36.4-65.7,97.5-86.9c50-17.4,115.2-27.1,183.4-27.1c67,0,131.3,9.4,181,26.6c29.1,10,53.6,23,71,37.4
	c18.9,15.8,28.9,33.1,28.9,50c0,34.5-40.7,69.4-106.3,91.1C328.6,507.1,266.6,515.6,200.5,515.6z M200.5,310.6
	c-64.7,0-128.7,9.4-175.5,25.7c-40.5,14.1-81.4,39.5-81.4,64.3c0,10,7.4,21.8,21,33.4c15.8,13.5,38.4,25.6,65.5,34.8
	c43.6,14.9,102.6,22.8,170.4,22.8c63.6,0,122.9-8,167-22.7c61.7-20.5,89.9-49.8,89.9-68.3c0-9.5-7.2-20.7-20.3-31.6
	c-15.1-12.6-37.1-24.1-63.4-33.2C326.4,319.6,264.9,310.6,200.5,310.6z"/>
<path class="st0" d="M86.6,650.8c-10.2,0-19.2-2.2-26.8-6.6C31,627.6,21.1,579.8,33.1,516.3c9.9-52.1,34.1-113.3,68.2-172.4h0
	c33.5-58,73.7-109,113.4-143.5c23.2-20.2,46.7-35,67.9-42.8c23.1-8.5,43.1-8.5,57.7-0.1c29.9,17.2,39.8,70,25.8,137.6
	c-9.9,48-33.5,105.9-66.5,163.2c-35.2,61-73.2,110.2-109.9,142.3c-23.8,20.8-48.3,36-70.7,43.9C107.4,648.7,96.4,650.8,86.6,650.8z
	 M122.1,355.9c-32.3,56-56.2,116.1-65.4,164.9c-8,42.1-6.5,90.2,15,102.6c16.1,9.3,55,0.5,102.3-40.9c34.7-30.3,71-77.5,104.9-136.2
	c31.8-55.1,54.4-110.5,63.8-156c6-29.1,7.3-55.5,3.8-76.4c-3-17.9-9.4-30.6-18.2-35.6c-8.2-4.7-21.5-4.1-37.5,1.8
	c-18.5,6.8-39.4,20.1-60.4,38.4C192.8,251.3,154.3,300.1,122.1,355.9L122.1,355.9z"/>
<path class="st0" d="M314.4,651.1c-27.2,0-61.7-16.4-97.7-47.4c-40.2-34.6-81.1-86.1-115.3-145.2v0c-33.6-58-57.6-118.3-67.7-170
	c-5.9-30.2-7-57.9-3.2-80.2c4.2-24.3,14.1-41.6,28.8-50.1c29.8-17.3,80.5,0.5,132.1,46.4c36.6,32.5,75,81.9,108.1,139.1
	c35.3,61,59,118.5,68.4,166.3c6.1,31,7.1,59.8,2.8,83.2c-4.6,24.9-15,42.6-30,51.3C333.2,648.9,324.3,651.1,314.4,651.1z
	 M122.2,446.5c32.4,56,72.6,106.7,110.2,139c32.5,28,74.9,50.6,96.4,38.2c16.1-9.3,27.9-47.4,15.7-109
	c-9-45.2-31.7-100.2-65.6-158.9c-31.9-55.1-68.6-102.4-103.3-133.2c-22.2-19.7-44.4-34.1-64.2-41.5c-17-6.4-31.2-7.1-39.9-2.1
	c-8.2,4.7-14.3,16.6-17.2,33.4c-3.3,19.4-2.3,44.2,3.1,71.5C66.9,333,89.9,390.7,122.2,446.5L122.2,446.5z"/>
</svg>
