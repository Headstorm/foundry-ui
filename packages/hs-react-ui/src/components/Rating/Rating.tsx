import React, { useState, useEffect, useRef } from 'react';
import UnstyledIcon from '@mdi/react';
import {
  mdiStar,
  mdiStarHalfFull,
  mdiStarOutline,
  mdiLockCheck,
  mdiLockOpenVariantOutline,
} from '@mdi/js';
import styled, { StyledComponentBase } from 'styled-components';
import { useTheme } from '../../context';
import variants from '../../enums/variants';
import { Div, Span } from '../../htmlElements';
import {
  getFontColorFromVariant,
  getBackgroundColorFromVariant,
  disabledStyles,
} from '../../utils/color';
import { SubcomponentPropsType } from '../commonTypes';
import { getShadowStyle } from '../../utils/styles';
import { mergeRefs } from '../../utils/refs';

export const Container: string & StyledComponentBase<any, {}, RatingContainerProps> = styled(Span)`
  ${({ elevation = 0, color, variant, disabled }: RatingContainerProps) => {
    const { colors } = useTheme();
    const backgroundColor = getBackgroundColorFromVariant(variant, color, colors.transparent);
    const fontColor = getFontColorFromVariant(variant, color, colors.background, colors.grayDark);

    return `
      font-size: 1em;
      border-radius: 0.25em;
      ${getShadowStyle(elevation, colors.shadow)}
      outline: 0 none;
      border: ${variant === variants.outline ? `1px solid ${color || colors.grayDark};` : '0 none;'}
      background-color: ${backgroundColor};
      color: ${fontColor};
      align-items: center;
      ${disabled ? disabledStyles() : ''}
    `;
  }}
`;

export const RatingWrapper = styled(Span)`
  ${() => {
    return `
      display: inline-flex;
      -webkit-flex-direction: row-reverse; 
      flex-direction: row-reverse;
      font-size: 1em;
      border-radius: 0.25em;
      outline: 0 none;
      align-items: center;
    `;
  }}
`;

const FilledRank = styled(Span)`
  ${({ color, variant }) => {
    const { colors } = useTheme();
    const backgroundColor = getBackgroundColorFromVariant(variant, color, colors.transparent);
    const fontColor = getFontColorFromVariant(variant, color, colors.background, colors.grayDark);
    return `
    padding: 1rem 0.5rem;
    background-color: ${backgroundColor};
    background-clip: text;
    // color: ${fontColor};
  `;
  }}
`;

const HalfRankContainer = styled(Span)`
  ${() => {
    return `
    padding: 1rem 0.5rem;
  `;
  }}
`;

const HalfFilledRank = styled(Div)`
  ${({ color, variant, hasHalfFilledRank }) => {
    const { colors } = useTheme();
    const fontColor = getFontColorFromVariant(variant, color, colors.background, colors.grayDark);
    return `
    position: absolute;
    color: ${fontColor};
    ${hasHalfFilledRank || 'clip-path: polygon(0% 0%, 50% 0%, 50% 100%, 0% 100%);'}
  `;
  }}
`;

const HalfUnfilledRank = styled(Div)`
  ${({ hasHalfFilledRank }) => {
    return `
    ${
      hasHalfFilledRank
        ? `content: none;`
        : `opacity: 50%;
        filter: saturate(0);
        clip-path: polygon(50% 0%, 100% 0%, 100% 100%, 50% 100%);`
    }
  `;
  }}
`;

const EmptyRank = styled(Div)`
  ${({ color, variant, hasStages, ratingSelected, isHoverable = false }: EmptyRankProps) => {
    const { colors } = useTheme();
    const fontColor = getFontColorFromVariant(variant, color, colors.background, colors.grayDark);

    return `
    padding: 1rem 0.5rem;
    ${
      hasStages
        ? `opacity: 50%;
       filter: saturate(0);
       ${
         !ratingSelected
           ? `&:hover,
       &:hover ~ div {
        opacity: 60%;
        filter: saturate(1);
       }`
           : ''
       }`
        : `opacity: 50%;
      filter: saturate(0);
      ${
        !ratingSelected && isHoverable
          ? `&:hover,
        &:hover ~ div {
          mix-blend-mode: multiply;
          color: ${fontColor};
          opacity: 100%;
          filter: saturate(1);
      }`
          : ''
      }`
    }
  `;
  }}
`;

const Info = styled(Div)`
  ${() => {
    return `
    position: relative;  
    text-align: right;  
    vertical-align: bottom;
    align-items: center;
    top: -1.3rem;        
    opacity: 60%;
    padding: 0px 1px;
  `;
  }}
`;

export type RatingContainerProps = {
  elevation: number;
  color: string;
  variant: variants;
  disabled: boolean;
};

export type EmptyRankProps = {
  color: string;
  variant: variants;
  ratingSelected: boolean;
  hasStages: boolean;
  isHoverable: boolean;
};

export type RatingProps = {
  value?: number;
  max?: number;
  onClick: (...args: any[]) => void;
  disabled?: boolean;
  elevation?: number;
  variant?: variants;
  color?: string;
  id?: string;
  testId?: string;

  stages?: Array<JSX.Element | string>;
  fractionStep?: number;
  filledRank?: string | JSX.Element;
  halfFilledRank?: string | JSX.Element;
  emptyRank?: string | JSX.Element;
  showDisplay?: boolean;

  containerProps?: SubcomponentPropsType;
  filledRankProps?: SubcomponentPropsType;
  halfFilledRankProps?: SubcomponentPropsType;
  emptyRankProps?: SubcomponentPropsType;

  containerRef?: React.RefObject<HTMLSpanElement>;

  StyledContainer?: string & StyledComponentBase<any, {}>;
  StyledFilledRankContainer?: string & StyledComponentBase<any, {}>;
  StyledHalfRankContainer?: string & StyledComponentBase<any, {}>;
  StyledEmptyRankContainer?: string & StyledComponentBase<any, {}>;
  StyledInfo?: string & StyledComponentBase<any, {}>;
};

const Rating = ({
  value = 1,
  max = 5,
  onClick,
  disabled = false,
  color,
  id,
  testId,
  elevation,
  variant = variants.fill,

  stages,
  fractionStep = 1,
  filledRank,
  halfFilledRank,
  emptyRank,
  showDisplay = false,

  StyledContainer = Container,
  StyledFilledRankContainer = FilledRank,
  StyledHalfRankContainer = HalfRankContainer,
  StyledEmptyRankContainer = EmptyRank,
  StyledInfo = Info,

  containerProps = {},
  filledRankProps = {},
  halfFilledRankProps = {},
  emptyRankProps = {},

  containerRef,

  ...nativeHTMLAttributes
}: RatingProps): JSX.Element => {
  let filledRankItem = filledRank || mdiStar;
  let halfFilledRankItem = halfFilledRank || mdiStarHalfFull;
  let emptyRankItem = emptyRank || mdiStarOutline;
  if (!filledRankItem) {
    // If a filledRank is not provided use default items, regardless of halfFilledRank and emptyRank's existence.
    filledRankItem = mdiStar;
    halfFilledRankItem = mdiStarHalfFull;
    emptyRankItem = mdiStarOutline;
  } else if (filledRank && !emptyRank) {
    // If filledRank is provided, but emptyRank is not, use the filledRank for all, disregarding existing halfFillRank.
    filledRankItem = filledRank;
    halfFilledRankItem = filledRank;
    emptyRankItem = filledRank;
  } else if (filledRank && !halfFilledRank && emptyRank) {
    // Use emptyRank for missing halfFilledRank
    filledRankItem = filledRank;
    halfFilledRankItem = emptyRank;
    emptyRankItem = emptyRank;
  }

  const { colors } = useTheme();
  const containerColor = color || colors.grayLight;
  const hasStages = Boolean(stages);
  const containerInternalRef = useRef();
  const hasHalfFilledRank = Boolean((halfFilledRank && emptyRank) || !filledRank);
  const [rawRating, setRawRating] = useState(0);

  const processRating = (rating: number) => {
    const step = fractionStep <= 1 && fractionStep > 0 ? fractionStep : 1;
    const inv: number = 1.0 / step;
    const usingFractions = Boolean(fractionStep !== 1);
    const newVal = usingFractions ? Math.round(rating * inv) / inv : Math.ceil(rating);
    return newVal <= max ? newVal : max;
  };

  // ratingValue is the returned value of the Rating component
  const [ratingValue, setRatingValue] = useState(processRating(value));
  // preHoverRatingValue is a temporary storage for the last selected ratingValue while the user hovers over the rating,
  // if a rating is not selected before the hover ends, the Rating component's ratingValue is reset using the preHoverRatingValue.
  const [preHoverRatingValue, setPreHoverRatingValue] = useState(processRating(value));

  // isRatingSelected is used to disable hovering styles if a ratingValue is selected or if the component is disabled
  const [isRatingSelected, _setRatingSelected] = useState(false);
  const isRatingSelectedRef = useRef(isRatingSelected);
  const setRatingSelected = (v: boolean) => {
    isRatingSelectedRef.current = v;
    _setRatingSelected(v);
  };
  const [preDisableRatingSelected, setPreDisableRatingSelected] = useState(disabled);

  const ratingListener = (ev: React.MouseEvent | Event) => {
    if (ev && ev.currentTarget) {
      const event = ev as React.MouseEvent;
      const node = event.currentTarget as HTMLElement;
      const rect = node.getBoundingClientRect();
      if (!isRatingSelectedRef.current) {
        setRatingValue(processRating(((event.clientX - rect.left) / rect.width) * max));
        setRawRating(((event.clientX - rect.left) / rect.width) * max);
      }
    }
  };
  // get everything we expose + anything consumer wants to send to container
  const mergedContainerProps = {
    id,
    elevation,
    color: containerColor,
    variant,
    ...containerProps,
  };

  const clickHandler = (e: React.MouseEvent) => {
    e.preventDefault();
    setPreHoverRatingValue(ratingValue);
    setRatingSelected(!isRatingSelected);
    onClick(ratingValue);
  };

  const mouseLeaveHandler = (e: React.MouseEvent) => {
    e.preventDefault();
    setRatingValue(preHoverRatingValue);
  };

  useEffect(() => {
    const current: HTMLElement | null = containerInternalRef
      ? ((containerInternalRef.current as unknown) as HTMLElement)
      : null;
    if (current) {
      current.addEventListener('mousemove', ratingListener);
    }

    if (disabled) {
      setPreDisableRatingSelected(isRatingSelected);
      setRatingSelected(true);
    } else {
      setRatingSelected(preDisableRatingSelected);
      setRatingValue(processRating(value));
      setPreHoverRatingValue(processRating(value));
    }
    return () => {};
  }, [value, fractionStep, max, showDisplay, disabled]);

  function rankContent(item: JSX.Element | string): JSX.Element {
    if (typeof item === 'string') {
      return <UnstyledIcon path={item} size="2rem" />;
    }
    return item;
  }

  const ratings: JSX.Element[] = [];

  if (stages) {
    for (let x = stages.length; x > 0; x--) {
      if (x <= Math.floor(ratingValue)) {
        ratings.push(
          <StyledFilledRankContainer onClick={clickHandler} key={x} {...filledRankProps}>
            {rankContent(stages[x - 1])}
          </StyledFilledRankContainer>,
        );
      } else {
        ratings.push(
          <StyledEmptyRankContainer
            hasStages={hasStages}
            ratingSelected={isRatingSelected}
            onClick={clickHandler}
            key={x}
            {...emptyRankProps}
          >
            {rankContent(stages[x - 1])}
          </StyledEmptyRankContainer>,
        );
      }
    }
  } else {
    // No rating stages. Use filled, halfFilled, and empty ratings
    for (let x = max; x > 0; x--) {
      if (x <= Math.round(ratingValue - 0.01)) {
        ratings.push(
          <StyledFilledRankContainer onClick={clickHandler} id={x} key={x} {...filledRankProps}>
            {rankContent(filledRankItem)}
          </StyledFilledRankContainer>,
        );
      } else if (x - 1 === Math.floor(ratingValue) && x - 1 !== ratingValue) {
        ratings.push(
          <StyledHalfRankContainer key={x}>
            <HalfFilledRank
              onClick={clickHandler}
              hasHalfFilledRank={hasHalfFilledRank}
              {...halfFilledRankProps}
            >
              {rankContent(halfFilledRankItem)}
            </HalfFilledRank>
            <HalfUnfilledRank
              onClick={clickHandler}
              hasHalfFilledRank={hasHalfFilledRank}
              {...halfFilledRankProps}
            >
              {rankContent(halfFilledRankItem)}
            </HalfUnfilledRank>
          </StyledHalfRankContainer>,
        );
      } else {
        ratings.push(
          <StyledEmptyRankContainer
            onClick={clickHandler}
            ratingSelected={isRatingSelected}
            key={x}
            isHoverable={Boolean(rawRating - x > 0 && rawRating - x < fractionStep / 2)}
            {...emptyRankProps}
          >
            {rankContent(emptyRankItem)}
          </StyledEmptyRankContainer>,
        );
      }
    }
  }

  return (
    <StyledContainer
      disabled={disabled}
      data-test-id={['hs-ui-rating', testId].join('-')}
      {...mergedContainerProps}
      {...nativeHTMLAttributes}
    >
      <RatingWrapper
        onMouseLeave={mouseLeaveHandler}
        value={ratingValue}
        ref={mergeRefs([containerRef, containerInternalRef])}
      >
        {ratings.map(ratingItem => ratingItem)}
      </RatingWrapper>
      {showDisplay && (
        <StyledInfo>
          <span style={{ marginRight: '2px', verticalAlign: 'middle' }}>{ratingValue}</span>
          <span>
            {isRatingSelectedRef.current ? (
              <UnstyledIcon style={{ verticalAlign: 'middle' }} path={mdiLockCheck} size="1rem" />
            ) : (
              <UnstyledIcon
                style={{ verticalAlign: 'middle' }}
                path={mdiLockOpenVariantOutline}
                size="1rem"
              />
            )}
          </span>{' '}
        </StyledInfo>
      )}
    </StyledContainer>
  );
};

Rating.Container = Container;
Rating.FilledRankContainer = FilledRank;
Rating.HalfRankContainer = HalfRankContainer;
Rating.EmptyRankContainer = EmptyRank;
Rating.Info = Info;
export default Rating;
