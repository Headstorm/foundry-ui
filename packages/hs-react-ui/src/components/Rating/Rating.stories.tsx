/* eslint-disable jsx-a11y/accessible-emoji */
import React, { useState } from 'react';
import { action } from '@storybook/addon-actions';
import { boolean, color, number, select } from '@storybook/addon-knobs';
import * as Icons from '@mdi/js';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import Rating from './Rating';
import colors from '../../enums/colors';
import variants from '../../enums/variants';
import { Span } from '../../htmlElements';

const StyledSpan = styled(Span)`
  font-size: 2rem;
`;

const options = {
  none: '',
  anvil: Icons.mdiAnvil,
  heart: Icons.mdiHeart,
  shield: Icons.mdiShield,
  circle: Icons.mdiCircle,
  crown: Icons.mdiCrown,
  cupCake: Icons.mdiCupcake,
  hamburger: Icons.mdiHamburger,
  mushroom: Icons.mdiMushroom,
  pig: Icons.mdiPigVariant,
  sofa: Icons.mdiSofa,
};

const halfOptions = {
  none: '',
  heart: Icons.mdiHeartHalfFull,
  star: Icons.mdiStarHalfFull,
  shield: Icons.mdiShieldHalfFull,
  circle: Icons.mdiCircleHalfFull,
};

const emptyOptions = {
  none: '',
  heart: Icons.mdiHeartOutline,
  star: Icons.mdiStarOutline,
  shield: Icons.mdiShieldOutline,
  circle: Icons.mdiCircleOutline,
  sofaOutline: Icons.mdiSofaOutline,
};

const design = {
  type: 'figma',
  url: 'https://www.figma.com/file/3r2G00brulOwr9j7F6JF59/Generic-UI-Style?node-id=105%3A5',
};

storiesOf('Rating', module)
  .addParameters({ component: Rating })
  .add(
    'Basic Rating',
    () => {
      const [val, setVal] = useState(1);
      const stepValue = number('fractionStep', 0.5, {
        range: false,
        min: 0.25,
        max: 1,
        step: 0.25,
      });
      return (
        <Rating
          value={number('value', val)}
          max={number('max', 5)}
          fractionStep={stepValue}
          variant={select('variant', variants, variants.text)}
          color={color('color', 'rgba(244,184,24,1)')}
          onClick={(newVal: number) => {
            setVal(newVal);
            action('onClick')(newVal);
          }}
          disabled={boolean('disabled', false)}
          elevation={number('elevation', 0)}
          filledRank={select('filledRank', options, '')}
          halfFilledRank={select('halfFilledRank', halfOptions, '')}
          emptyRank={select('emptyRank', emptyOptions, '')}
          showDisplay={boolean('showDisplay', true)}
        />
      );
    },
    { design },
  )
  .add(
    'Stages Rating',
    () => {
      const [val, setVal] = useState(1);
      const stages = [
        <StyledSpan role="img" aria-label="mad-rating">
          😡
        </StyledSpan>,
        <StyledSpan role="img" aria-label="not-happy-rating">
          😞
        </StyledSpan>,
        <StyledSpan role="img" aria-label="meh-rating">
          😑
        </StyledSpan>,
        <StyledSpan role="img" aria-label="glad-rating">
          😁
        </StyledSpan>,
        <StyledSpan role="img" aria-label="geeked-out-rating">
          🤓
        </StyledSpan>,
      ];

      return (
        <Rating
          variant={select('variant', variants, variants.text)}
          color={color('color', colors.primaryDark)}
          onClick={(newVal: number) => {
            setVal(newVal);
            action('onClick')(newVal);
          }}
          stages={stages}
          disabled={boolean('disabled', false)}
          elevation={number('elevation', 1)}
          value={number('value', val)}
          showDisplay={boolean('showDisplay', true)}
        />
      );
    },
    { design },
  );
