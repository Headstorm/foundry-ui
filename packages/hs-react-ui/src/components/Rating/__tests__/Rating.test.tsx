import React from 'react';
import { render, configure, waitFor } from '@testing-library/react';
import Rating from '../Rating';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);
configure({ testIdAttribute: 'data-test-id' });

const testId = 'foundry-rating';

describe('Rating', () => {
  describe('Accessibility Tests', () => {
    it('Should pass accessibility test with default props', async () => {
      //
      const component = (
        <Rating onClick={() => {}} containerProps={{ 'aria-label': 'aria-label-test' }}></Rating>
      );
      const { container } = render(component);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });
  it('shows Rating with default props', async () => {
    const { container, getByTestId } = render(
      <Rating onClick={() => {}} containerProps={{ 'data-test-id': testId }} />,
    );
    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });
  describe('Ref tests', () => {
    it('containerRef.current should exist', async () => {
      const ref = React.createRef<HTMLSpanElement>();
      const { getByTestId } = render(
        <Rating
          onClick={() => {}}
          containerProps={{ 'data-test-id': testId }}
          containerRef={ref}
        />,
      );
      await waitFor(() => getByTestId(testId));
      console.log(ref);
      expect(ref.current instanceof HTMLSpanElement).toBeTruthy();
    });
  });
});
