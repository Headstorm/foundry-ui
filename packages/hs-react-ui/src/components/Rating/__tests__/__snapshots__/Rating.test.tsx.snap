// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating shows Rating with default props 1`] = `
.c0 {
  font-size: 1em;
  border-radius: 0.25em;
  outline: 0 none;
  border: 0 none;
  background-color: #7c8186;
  color: #000;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row-reverse;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  font-size: 1em;
  border-radius: 0.25em;
  outline: 0 none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  padding: 1rem 0.5rem;
  background-color: undefined;
  background-clip: text;
}

.c2 {
  padding: 1rem 0.5rem;
  opacity: 50%;
  -webkit-filter: saturate(0);
  filter: saturate(0);
}

<div>
  <span
    class=" c0"
    color="#7c8186"
    data-test-id="foundry-rating"
  >
    <span
      class=" c1"
      value="1"
    >
      <div
        class=" c2"
      >
        <svg
          role="presentation"
          style="width: 2rem; height: 2rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
            style="fill: currentColor;"
          />
        </svg>
      </div>
      <div
        class=" c2"
      >
        <svg
          role="presentation"
          style="width: 2rem; height: 2rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
            style="fill: currentColor;"
          />
        </svg>
      </div>
      <div
        class=" c2"
      >
        <svg
          role="presentation"
          style="width: 2rem; height: 2rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
            style="fill: currentColor;"
          />
        </svg>
      </div>
      <div
        class=" c2"
      >
        <svg
          role="presentation"
          style="width: 2rem; height: 2rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
            style="fill: currentColor;"
          />
        </svg>
      </div>
      <span
        class=" c3"
        id="1"
      >
        <svg
          role="presentation"
          style="width: 2rem; height: 2rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
            style="fill: currentColor;"
          />
        </svg>
      </span>
    </span>
  </span>
</div>
`;
